fixed reading/writing logic and aggregation

kimkao opened the pull request:
Fixed process_board and bom_automator:

Added aggregation, initially forgot to consider the board quantities. Some bugs after the commit with just the aggregation: I realized that my previous code could potentially read a board file before first initializing board_qty dictionary, which would cause an error when trying to process the board file. I fixed this by implementing two for loops - the first for extracting board_qty.csv first, then the second for reading all the board files. Additionally, I added a lineterminator argument for the csv writer constructor in generate_part_number_csv and generate_url_csv to get rid of the extra rows in the outputted csv files.

kimkao committed:
fixed reading/writing logic
Fixed process_board and bom_automator:

Added aggregation. Some bugs after the commit with just the aggregation: I realized that my previous code could potentially read a board file before first initializing board_qty dictionary, which would cause an error when trying to process the board file. I fixed this by implementing two for loops - the first for extracting board_qty.csv first, then the second for reading all the board files. Additionally, I added a lineterminator argument for the csv writer constructor in generate_part_number_csv and generate_url_csv to get rid of the extra rows in the outputted csv files.

krystynneisess commented:
I've added some comments to commits 6022b087ddac6d17151594f200a31cef96ed7918 and a3aa371b77447472e51f546b1318c4ddb5efa2fe.  Do take a look!

kimkao committed:
Added some checks for file reading
Instead of just having users pass in files when running bom_automator Main, we now will print instructions for how to format each file, then instruct the user to pass in files in the correct format. Also, made all the board names lower case when reading them from the csv files and when storing them in board_qty.

krystynneisess commented:
Will be resolved in subsequent pull request!
